pkg/client/cache/eventqueue_test.go

```diff
@@ -217,6 +217,24 @@ func TestEventQueue_PopAfterResyncShouldBeTypeModified(t *testing.T) {
 	}
 }
 
+func TestEventQueue_ResyncsShouldNotCausePanics(t *testing.T) {
+	q := NewEventQueue(keyFunc)
+	q.Add(cacheable{"foo", 10})
+	q.Pop()
+
+	q.Resync()
+
+	q.Delete(cacheable{key: "foo"})
+
+	q.Pop()
+
+	items := q.List()
+
+	if len(items) > 0 {
+		t.Fatalf("expected the list to be empty after a coalesce, but it was not")
```

Comment:
When I wrote this code, I'm not sure what I was getting at...

How about: "expected Resync() not to add a duplicate item to the event queue, but it did"